src/noit_metric_tag_search.c

```diff
@@ -1255,3 +1255,34 @@ noit_metric_tag_search_unparse(const noit_metric_tag_search_ast_t *search) {
   mtev_dyn_buffer_destroy(&buf);
   return res;
 }
+
+int
+noit_metric_tag_search_swap(noit_metric_tag_search_ast_t *search, int idx1, int idx2) {
+  if (!search) {
+    return -1;
+  }
+  if (idx1 < 0 || idx2 < 0) {
+    return -1;
+  }
+  int nargs = noit_metric_tag_search_get_nargs(search);
+  if (nargs == 0) {
+    return -1;
+  }
```

Comment:
This == 0 is redundant to the next predicate

Reply:
Fixed